packages/adblocker/src/resources.ts

```diff
@@ -20,14 +22,86 @@ function btoaPolyfill(buffer: string): string {
   return buffer;
 }
 
-export interface Resource {
+export interface Redirect {
   contentType: string;
   body: string;
 }
 
-// TODO - support # alias
+type ResourceBase = {
+  names: string[];
+  content: string;
+};
+
+export type Resource = ResourceBase & {
+  contentType: string;
+  encoding?: 'base64';
+};
+
+export type Scriptlet = ResourceBase & {
+  dependencies: string[];
+  fnName: string;
+  executionWorld: Scripting.ExecutionWorld;
+  requiresTrust: boolean;
+};
+
+type PartialBy<T, K extends keyof T> = Pick<Partial<T>, K> & Omit<T, K>;
+
+export type ResourcesDistribution = {
+  redirects: Resource[];
```

Comment:
I don't understand why this is not Redirect and why do we need both Resource and Redirect types

Reply:
Redirect type is a corresponding type for Firefox redirect details not our internals.